src/build/mod.rs

```diff
@@ -167,33 +141,35 @@ pub struct SourceMetadata {
 impl BuildContext {
     pub fn new(
         cache_dir: PathBuf,
-        dot_pixi_dir: PathBuf,
         channel_config: ChannelConfig,
         variant_config: Targets<Option<HashMap<String, Vec<String>>>>,
         tool_context: Arc<ToolContext>,
+        command_dispatcher: CommandDispatcher,
     ) -> Result<Self, std::io::Error> {
         Ok(Self {
             channel_config,
             glob_hash_cache: GlobHashCache::default(),
             source_metadata_cache: SourceMetadataCache::new(cache_dir.clone()),
             build_cache: BuildCache::new(cache_dir.clone()),
-            cache_dir,
-            work_dir: dot_pixi_dir.join("build-v0"),
+            work_dir: command_dispatcher.cache_dirs().working_dirs(),
```

Comment:
This changed from `build-v0` to `work-v0` both names are not really saying much. Can we think of something that helps users understand what could possibly be in these directories. 

`build-work-v0` would already be better but maybe you can come up with an even better one! 🙏 

ps. does this potentially create strange behavior for the users that we have to take care of? 

ps2. please add a `pixi clean cache --build` so that it removes all build related caches like this one.

Reply:
That was unintentional, lets rename that to what it was and figure out a better name later.

Same goes for cleaning. If thats not there currently I think its out of scope for this Pr.

Reply:
@Hofer-Julian is working on the cache dir fixes. 

These other ps's can go into an issue, they're not holding back this PR 👍